src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs

```diff
@@ -198,49 +198,23 @@ internal void MarkHandshakeCompleted()
         public static SafeSslHandle Create(SafeSslContextHandle context, bool isServer)
         {
             SafeBioHandle readBio = Interop.Crypto.CreateMemoryBio();
-            if (readBio.IsInvalid)
-            {
-                return new SafeSslHandle();
-            }
-
             SafeBioHandle writeBio = Interop.Crypto.CreateMemoryBio();
-            if (writeBio.IsInvalid)
-            {
-                readBio.Dispose();
-                return new SafeSslHandle();
-            }
-
             SafeSslHandle handle = Interop.Ssl.SslCreate(context);
-            if (handle.IsInvalid)
+            if (readBio.IsInvalid || writeBio.IsInvalid || handle.IsInvalid)
             {
                 readBio.Dispose();
                 writeBio.Dispose();
+                handle.Dispose(); // will make IsInvalid==true if it's not already
                 return handle;
             }
             handle._isServer = isServer;
 
-            // After SSL_set_bio, the BIO handles are owned by SSL pointer
-            // and are automatically freed by SSL_free. To prevent a double
-            // free, we need to keep the ref counts bumped up till SSL_free
-            bool gotRef = false;
-            readBio.DangerousAddRef(ref gotRef);
-            try
-            {
-                bool ignore = false;
-                writeBio.DangerousAddRef(ref ignore);
-            }
-            catch
-            {
-                if (gotRef)
-                {
-                    readBio.DangerousRelease();
-                }
-                throw;
-            }
-
-            Interop.Ssl.SslSetBio(handle, readBio, writeBio);
+            // SslSetBio will transfer ownership of the BIO handles to the SSL context
+            readBio.TransferOwnershipToParent(handle);
+            writeBio.TransferOwnershipToParent(handle);
             handle._readBio = readBio;
             handle._writeBio = writeBio;
+            Interop.Ssl.SslSetBio(handle, readBio, writeBio);
 
```

Comment:
"very bad state" here would at worst I think mean a leaked BIO handle.  I'm not sure that's worth fail fasting for, but since it shouldn't happen anyway, I could add a fail fast in a catch block I think with little harm.